src/__tests__/resolver-test.ts

```diff
@@ -291,33 +363,39 @@ describe('resolver', () => {
         )
 
         await expect(resolver.resolve('did:mock:abcdef')).resolves.toEqual({
-          '@context': 'https://w3id.org/did/v1',
-          id: 'did:mock:abcdef',
-          publicKey: [
-            {
-              id: 'owner',
-              controller: '1234',
-              type: 'xyz'
-            }
-          ]
+          didResolutionMetadata: { contentType: 'application/did+json' },
+          didDocument: {
+            id: 'did:mock:abcdef',
+            verificationMethod: [
+              {
+                id: 'owner',
+                controller: '1234',
+                type: 'xyz'
+              }
+            ]
+          },
+          didDocumentMetadata: {}
         })
         await expect(
-          resolver.resolve('did:mock:abcdef;no-cache=true')
+          resolver.resolve('did:mock:abcdef?versionId=2')
         ).resolves.toEqual({
-          '@context': 'https://w3id.org/did/v1',
-          id: 'did:mock:abcdef',
-          publicKey: [
-            {
-              id: 'owner',
-              controller: '1234',
-              type: 'xyz'
-            }
-          ]
+          didResolutionMetadata: { contentType: 'application/did+json' },
+          didDocument: {
+            id: 'did:mock:abcdef',
+            verificationMethod: [
+              {
+                id: 'owner',
+                controller: '1234',
+                type: 'xyz'
+              }
+            ]
+          },
+          didDocumentMetadata: {}
         })
         return expect(mockmethod).toBeCalledTimes(2)
       })
 
-      it('should not cache null docs', async () => {
+      it.skip('should not cache null docs', async () => {
```

Comment:
I think that a test like this should not be skipped.

Reply:
See my question from the first post:

> Question about the last test: Why do we not want to cache null results, i.e. notFound? Seems to me that we should cache that just the same way we cache any other result?
Skipped this test for now.

Any thoughts on this`

Reply:
Because `null` doc can be caused by a temporary error.
Consider `did:web` where a 404 error on the `.well-known/did.json` might be just a glitch.